docs/source/transaction_family_specifications/track_and_trade_transaction_family.rst

```diff
@@ -0,0 +1,747 @@
+        sint64 latitude = 1;
+        sint64 longitude = 2;
+  }
+
+
+Agents
+------
+
+Agents are entities that can send transactions affecting Records. This
+could include not only humans and companies that act as owners and
+custodians of objects being tracked, but also autonomous sensors
+sending transactions updating Records' data. All Agents must be
+registered on-chain before interacting with Records.
+
+.. code-block:: protobuf
+
+    message Agent {
+        // The Agent's public key. This must be unique.
+        string public_key = 1;
+
+        // A human-readable name identifying the Agent.
+        string name = 2;
+
+        // Approximately when the Agent was registered, as a Unix UTC timestamp
+        uint64 timestamp = 3;
+    }
+
+Agents whose keys have the same hash are stored in a list alphabetized
+by public key.
+
+.. code-block:: protobuf
+
+    message AgentContainer {
+        repeated Agent entries = 1;
+    }
+
+
+Proposals
+---------
+
+A Proposal is an offer from the owner or custodian of a Record to
+authorize another Agent as an owner, custodian, or reporter for that
+Record. Proposals are tagged as being for transfer of ownership,
+transfer of custodianship, or authorization of a reporter for some
+Properties. Proposals are also tagged as being open, accepted,
+rejected, or canceled. There cannot be more than one open Proposal for
+a specified role for each combination of Record, receiving Agent, and
+issuing Agent.
+
+.. code-block:: protobuf
+
+   message Proposal {
+       enum Role {
+           OWNER = 1;
+           CUSTODIAN = 2;
+           REPORTER = 3;
+       }
+
+       enum Status {
+           OPEN = 1;
+           ACCEPTED = 2;
+           REJECTED = 3;
+           CANCELED = 4;
+       }
+
+       // The id of the Record with which this Proposal deals
+       string record_id = 1;
+
+       // Approximately when this proposal was created, as a Unix UTC timestamp
+       uint64 timestamp = 2;
+
+       // The public key of the Agent that created the Proposal
+       string issuing_agent = 3;
+
+       // The public key of the Agent to which the Proposal is addressed
+       string receiving_agent = 4;
+
+       // Whether the Proposal is for transfer of ownership or
+       // custodianship or reporter authorization
+       Role role = 5;
+
+       // The names of properties for which the reporter is being authorized
+       // (empty for owner or custodian transfers)
+       repeated string properties = 6;
+
+       // Whether the Proposal is open, accepted, rejected, or canceled.
+       // For a given Record and receiving Agent, there can be only one
+       // open Proposal at a time for each role.
+       Status status = 7;
+
+       // human-readable terms of transfer
+       string terms = 8;
+   }
+
+
+Proposals with the same address are stored in a list sorted
+alphabetically first by record_id, then by receiving_agent, then by
```

Comment:
Double backticks needed for record_id, receiving_agent, and timestamp (I think).